documentation/modules/oauth/con-oauth-authorization-keycloak-example.adoc

```diff
@@ -0,0 +1,462 @@
+// Module included in the following module:
+//
+// con-oauth-authorization-keycloak-authorization-services.adoc
+
+[id='con-oauth-authorization-keycloak-example_{context}']
+= Example authorization rules configuration using Authorization Services
+
+This is an end-to-end example of using Keycloak Authorization Services to configure authorization rules for use with `keycloak` authorization in Strimzi.
+The example starts by deploying the Keycloak server with pre-configured realms, requiring no additional configuration.
+We deploy the Kafka cluster configured to use one of the pre-configured realms.
+We connect to the Keycloak Admin Console and demonstrate the sections of the user interface where authorizations rules are configured.
+We use Kafka CLI client tools with different personal and service accounts to demonstrate limited access based on the permissions granted to different accounts.
+
+
+== Token Based Authorization with Keycloak Authorization Services
+
+Once the Kafka Broker has obtained an access token by using `oauth` authentication, it is possible to use centrally managed authorization rules to enforce access restrictions onto Kafka Clients.
+For this, Strimzi Kafka Operator comes with `keycloak` authorization which uses Keycloak Authorization Services to centrally manage permissions.
+
+When using `keycloak` authorization a custom authorizer is configured on the Kafka Broker that uses Authorization Services REST endpoints available on Keycloak, which provide a list of granted permissions on resources for authenticated users.
+The list of grants (permissions) is fetched as the first action after an authenticated session is established by the Kafka client, and then regularly refreshed in the background.
+Grants are cached and enforced locally on the Kafka Broker for each user session to provide fast authorization decisions. Because they are refreshed, any changes to the grants on the Keycloak server are detected and enforced.
```

Comment:
```suggestion
Grants are cached and enforced locally on the Kafka broker for each user session to provide fast authorization decisions. Because they are refreshed, any changes to the grants on the Keycloak server are detected and enforced.
```